spyder/utils/programs.py

```diff
@@ -288,7 +287,10 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                                    "an external terminal"),
                                  QMessageBox.Ok)
     elif os.name == 'posix':
-        programs = [{'cmd': 'gnome-terminal',
+        programs = [{'cmd': 'x-terminal-emulator',
+                     'wdir-option': '--working-directory',
+                     'execute-option': '-x'},

```

Comment:
I put the test in if I remember correctly, but it just reflects the code as it was at the time, so feel free to change this (in another PR).